be/src/connector/hive_connector.cpp

```diff
@@ -2,24 +2,306 @@
 
 #include "connector/hive_connector.h"
 
+#include "exec/exec_node.h"
+#include "exec/vectorized/hdfs_scanner_orc.h"
+#include "exec/vectorized/hdfs_scanner_parquet.h"
+#include "exec/vectorized/hdfs_scanner_text.h"
+#include "exprs/expr.h"
+#include "storage/chunk_helper.h"
+
 namespace starrocks {
 namespace connector {
+using namespace vectorized;
+
+// ================================
+
+DataSourceProviderPtr HiveConnector::create_data_source_provider(vectorized::ConnectorScanNode* scan_node,
+                                                                 const TPlanNode& plan_node) const {
+    return std::make_unique<HiveDataSourceProvider>(scan_node, plan_node);
+}
+
+// ================================
+
+HiveDataSourceProvider::HiveDataSourceProvider(vectorized::ConnectorScanNode* scan_node, const TPlanNode& plan_node)
+        : _scan_node(scan_node), _hdfs_scan_node(plan_node.hdfs_scan_node) {}
+
+DataSourcePtr HiveDataSourceProvider::create_data_source(const TScanRange& scan_range) {
+    return std::make_unique<HiveDataSource>(this, scan_range);
+}
+
+// ================================
+
+HiveDataSource::HiveDataSource(const HiveDataSourceProvider* provider, const TScanRange& scan_range)
+        : _provider(provider), _scan_range(scan_range.hdfs_scan_range) {}
 
-Status HiveDataSource::do_open(RuntimeState* state) {
+Status HiveDataSource::init() {
```

Comment:
we can move all this to `HiveDataSourceProvider::create_data_source`

Reply:
Explained before.